src/opt/ImproveMemoryOps.cpp

```diff
@@ -188,14 +189,16 @@ static llvm::Value *lGetBasePtrAndOffsets(llvm::Value *ptrs, llvm::Value **offse
                         llvm::Value *zeroMask = llvm::ConstantVector::getSplat(
                             llvm::ElementCount::get(bop_var_type->getNumElements(), false),
                             llvm::Constant::getNullValue(llvm::Type::getInt32Ty(*g->ctx)));
-                        shuffle_offset = new llvm::ShuffleVectorInst(bop_var, llvm::UndefValue::get(bop_var_type),
-                                                                     zeroMask, "shuffle");
+                        shuffle_offset =
+                            new llvm::ShuffleVectorInst(bop_var, llvm::UndefValue::get(bop_var_type), zeroMask,
+                                                        "shuffle", ISPC_INSERTION_POINT_INSTRUCTION(bop_var));
```

Comment:
This instruction is inserted twice now.

Reply:
Good catch! That's an evil autocomplete in Cursor.

Reply:
I've started a trunk-rebuild workflow as well: https://github.com/ispc/ispc/actions/runs/11020210568

Reply:
Ok, CI fails are fixed, the benchmark fails are fix in a separate commit.

One more pipeline to verify that: https://github.com/ispc/ispc/actions/runs/11023557508